File: frontend/src/types/UserPlantDTO.ts
import type { PlantDTO } from "./PlantDTO";

export interface UserPlantDTO {
  id: string;
  nickname?: string;
  imageUrl?: string;
  wateringFrequencyDays?: number;
  createdAt?: string;
  diedAt?: string;
  plant: PlantDTO;
}

File: backend/src/main/java/com/happyplants/model/dto/UserPlantDTO.java
package com.happyplants.model.dto;

import java.io.Serializable;
import java.time.OffsetDateTime;
import java.util.UUID;

public record UserPlantDTO(
        UUID id,
        String nickname,
        String imageUrl,
        Integer wateringFrequencyDays,
        OffsetDateTime createdAt,
        OffsetDateTime diedAt,
        OffsetDateTime lastWateredAt,
        int timesWatered,
        PlantDTO plant
) implements Serializable {
}
